### DISCUSSION TITLE
Media preview not working with head tag in dynamic route

### DISCUSSION THREAD

**Discussion by mohsinghafoor1** [vercel/next.js#53907] [Help] [answered]
### Summary

`
import { Center, Container, Flex, Grid, Loader, rem } from "@mantine/core";
import { Fragment, useEffect, useState } from "react";
import Head from "next/head";
import { GetServerSideProps, InferGetServerSidePropsType } from "next";
import Header from "@/components/Header";
import BlogSection from "@/components/blog/Section";
import BlogArticleHeader from "@/components/blog/article/ArticleHeader";
import Footer from "@/components/Footer";
import Content from "@/components/blog/article/Content";
import { Blog } from "@/defs/Types";

async function fetchBlogData(id: string): Promise<Blog> {
  const response = await fetch(
    `url`,
    {
      headers: {
        Authorization: `Bearer`
      },
    }
  );

  const res = await response.json();
  return res.data;
}

export const getServerSideProps: GetServerSideProps = async (context) => {
  const { id } = context.query;
  const blogData = await fetchBlogData(id as string);

  return {
    props: {
      blogData,
    },
  };
};

export default function BlogArticle({
  blogData,
}: InferGetServerSidePropsType<typeof getServerSideProps>) {
  const [loading, setLoading] = useState<boolean>(true);
  const [blog, setBlog] = useState<Blog>(blogData);
  useEffect(() => {
    if (blogData) {
      setBlog(blogData);
      setLoading(false);
    }
  }, [blogData]);
  return (
      <Fragment>
        <Header />
        {loading && !blog && <Center w={'100%'} h={500}><Loader color='grey' size={50}/></Center>}
        {!loading && blog &&
          <BlogSection>
            <Head>
              <meta charSet="utf-8"></meta>
              <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"></meta>
              <meta name="viewport" content="width=device-width"></meta>
              <title>{`${blog.attributes.title.slice(0, 50)}...`}</title>
              <meta name="description" content={blog.attributes.description} />
              <meta property="og:title" content={`${blog.attributes.title.slice(0, 50)}...`} />
              <meta property="og:description" content={blog.attributes.description} />
              <meta property="og:type" content="website" />
              <meta property="twitter:title" content={`${blog.attributes.title.slice(0, 50)}...`} />
              <meta property="twitter:description" content={blog.attributes.description} />
              <meta property="twitter:card" content="summary_large_image"></meta>
              <meta property="og:image" content={blog.attributes.coverImage}/>
              <meta property="og:image:width" content="1200" />
              <meta property="twitter:image:width" content="1024" />
              <meta property="og:image:height" content="627" />
              <meta property="twitter:image" content={blog.attributes.coverImage} />
              <meta property="twitter:image:height" content="512" />
              <link rel="shortcut icon" href="/favicon.ico" />
            </Head>
            <Container size="lg">
              <BlogArticleHeader
                tags={blog.attributes.tags} 
                title={blog.attributes.title} 
                desc={blog.attributes.description} 
                author={`${blog.attributes.author.data.attributes.firstname} ${blog.attributes.author.data.attributes.lastname}`} 
                img={blog.attributes.coverImage} 
                subtext={blog.attributes.coverText} 
                id={blog.id} 
                slugUrl={blog.attributes.urlSlug}
              />
              <Grid grow gutter={rem(59)} mt={rem(16)} columns={18}>
                <Grid.Col span={13}>
                  <Flex gap={rem(35)} direction="column">
                  <Content content={blog.attributes.content}/>
                  </Flex>
                </Grid.Col>
              </Grid>
            </Container>
          </BlogSection>}
        <Footer />
      </Fragment>
  );
}

`

### Additional information

```js
I have this component in my next.js app and this is my dynamic route with path directory of /pages/blog/article/\[urlSlug\]/\[id\]/tsx
but when i paste my url in facebook or twitter my image title or description is not showing in preview.

I tried server side rendering as i have multiple blogs and i'm getting blogdata on bases of id.
```

**Comment by Nadeera3784** [comment, marked as answer]
Hi @mohsinghafoor1 ,
Are you using open graph tags? if not try to use [open graph tags](https://developers.facebook.com/docs/sharing/webmasters#markup) in your site markup to get rich link support in social media.
Additionally  you can try to fetch Metadata from the Server. for example :

```
export const getServerSideProps: GetServerSideProps = async (context) => {
  const { id } = context.query;
  const blogData = await fetchBlogData(id as string);

  return {
    props: {
      blogData,
      title: blogData.attributes.title,
      description: blogData.attributes.description,
      imageUrl: blogData.attributes.coverImage,
    },
  };
};

```

**Reply by mohsinghafoor1** [reply]
Thank you @Nadeera3784, this solution worked for me.
